test/k8sT/fqdn.go

```diff
@@ -0,0 +1,168 @@
+	var (
+		kubectl          *helpers.Kubectl
+		microscopeErr    error
+		microscopeCancel                    = func() error { return nil }
+		backgroundCancel context.CancelFunc = func() { return }
+		backgroundError  error
+
+		bindManifest = helpers.ManifestGet("bind_deployment.yaml")
+		demoManifest = helpers.ManifestGet("demo.yaml")
+
+		apps    = []string{helpers.App2, helpers.App3}
+		appPods map[string]string
+
+		worldTarget          = "http://world1.cilium.test"
+		worldTargetIP        = "192.168.9.10"
+		worldInvalidTarget   = "http://world2.cilium.test"
+		worldInvalidTargetIP = "192.168.9.11"
+	)
+
+	BeforeAll(func() {
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+		ProvisionInfraPods(kubectl)
+
+		bindManifest = helpers.ManifestGet("bind_deployment.yaml")
+
+		res := kubectl.Apply(bindManifest)
+		res.ExpectSuccess("Bind config cannot be deployed")
+
+		res = kubectl.Apply(demoManifest)
+		res.ExpectSuccess("Bind config cannot be deployed")
+
+		err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
+		Expect(err).Should(BeNil(), "Demo app is not ready after timeout")
+
+		appPods = helpers.GetAppPods(apps, helpers.DefaultNamespace, kubectl, "id")
+
+		err = kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=bind", 300)
+		Expect(err).Should(BeNil(), "Demo app is not ready after timeout")
+
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport(helpers.KubeSystemNamespace,
+			"cilium service list",
+			"cilium endpoint list")
+	})
+
+	AfterAll(func() {
+		_ = kubectl.Delete(bindManifest)
+		_ = kubectl.Delete(demoManifest)
+		ExpectAllPodsTerminated(kubectl)
+	})
+
+	JustBeforeEach(func() {
+		microscopeErr, microscopeCancel = kubectl.MicroscopeStart()
+		Expect(microscopeErr).To(BeNil(), "Microscope cannot be started")
+		backgroundCancel, backgroundError = kubectl.BackgroundReport("uptime")
+		Expect(backgroundError).To(BeNil(), "Cannot start background report process")
+	})
+
+	JustAfterEach(func() {
+		kubectl.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
+		Expect(microscopeCancel()).To(BeNil(), "cannot stop microscope")
+		backgroundCancel()
+	})
+
+	AfterEach(func() {
+		_ = kubectl.Exec(fmt.Sprintf("%s delete --all cnp", helpers.KubectlCmd))
+	})
+
+	It("Restart Cilium validate that FQDN is still working", func() {
+		connectivityTest := func() {
+			res := kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(worldTarget))
+			ExpectWithOffset(1, res).To(helpers.CMDSuccess(), "%q cannot curl to %q",
+				appPods[helpers.App2], worldTarget)
+
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(worldInvalidTarget))
+			ExpectWithOffset(1, res).ShouldNot(helpers.CMDSuccess(),
+				"%q can curl to %q when it should fail", appPods[helpers.App2], worldInvalidTarget)
+
+			By("Testing that connection using IP is working correctly")
```

Comment:
Update this `By` statement to indicate that app2 connection to the specific IP is working.